tests/unit/test_gbq.py

```diff
@@ -47,6 +47,24 @@ def mock_get_credentials(*args, **kwargs):
     return mock_credentials, "default-project"
 
 
+def mock_get_service_account_credentials(*args, **kwargs):
```

Comment:
Style nit: Let's add `@pytest.fixture` to this and rename to `mock_service_account_credentials`.

It doesn't need to take any arguments, and it only needs to return the `mock_credentials`. Same for `mock_compute_engine_credentials`.

Then, rather than calling these functions in the test, use them as a "pytest fixture" by adding a `mock_service_account_credentials` or `mock_compute_engine_credentials` argument to the test function(s).